gtsfm/data_association/dsf_tracks_estimator.py

```diff
@@ -42,17 +42,17 @@ def run(self, matches_dict: Dict[Tuple[int, int], np.ndarray], keypoints_list: L
             list of all valid SfmTrack2d generated by the matches.
         """
         # check to ensure dimensions of coordinates are correct
-        dims_valid = all([kps.coordinates.ndim == 2 for kps in keypoints_list])
-        if not dims_valid:
-            raise Exception("Dimensions for Keypoint coordinates incorrect. Array needs to be 2D")
+        # dims_valid = all([kps.coordinates.ndim == 2 for kps in keypoints_list])
```

Comment:
why is this removed?

Reply:
This doesn't allow for no keypoints. This was just a hack, but I could have it ignore None but check the others

Reply:
did you come across a case where we had no keypoints in an image? how is that handled? are Keypoints None? Or is it an empty matrix? 

`all([kps.coordinates.ndim == 2 for kps in keypoints_list if kps is not None])`

Reply:
Yes, I had a couple cases when using LoFTR. Keypoints.coordinates is None. I'll add this.

Reply:
added